transport/query.go

```diff
@@ -16,8 +16,8 @@ type Statement struct {
 	Compression       bool
 }
 
-func newQueryForStatement(s Statement, pagingState frame.Bytes) frame.Request {
-	return &Query{
+func makeQueryForStatement(s Statement, pagingState frame.Bytes) Query {
```

Comment:
@Choraden @Kulezi the same needs to be done for execute and maybe prepare.

In the context of Conn::Query there is an optiomisation for stack pointers, it's better to copy Query to stack and point to stack than allocate it on heap - no GC...